yt/data_objects/static_output.py

```diff
@@ -974,8 +974,12 @@ def _assign_unit_system(self, unit_system):
 
         us = create_code_unit_system(
```

Comment:
apparently this line needs to be executed even when `unit_system != "code"`, or else some tests 
(e.g. `yt/data_objects/tests/test_sph_data_objects.py::test_compare_arbitrary_grid_slice`) will fail. I'm not convinced the current behaviour isn't actually buggy...